src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sources/streamfactories/TwitterStreamFactory.scala

```diff
@@ -62,18 +62,17 @@ class TwitterStreamFactory(configurationManager: ConfigurationManager) extends S
       query = Some(query)
     )
 
-    def isOriginalTweet(status: Status) : Boolean = {
-      !status.isRetweet && status.getRetweetedStatus == null
-    }
-
     params.getOrElse("trustedSourceFilterEnabled", "true").toString.toBoolean match {
       case false => stream
       case true => {
-        val trustedSourceScreenNames = configurationManager.fetchTrustedSources(sparkContext = ssc.sparkContext)
-          .filter(source=>source.pipelinekey.equalsIgnoreCase("twitter"))
-          .map(source=>source.externalsourceid).toSet
-
-        stream.filter(status=>{ trustedSourceScreenNames.contains(status.getUser.getScreenName) && isOriginalTweet(status) })
+        val trustedSourceScreenNames = params.getTrustedSources.toSet
+        stream.filter(status=> {
+          def isOriginalTweet(status: Status) : Boolean = {
+            !status.isRetweet && status.getRetweetedStatus == null
+          }
+
+          trustedSourceScreenNames.contains(status.getUser.getScreenName) && isOriginalTweet(status)
```

Comment:
I was thinking, should we only do this `trustedSourcesScreenNames` check if the set is non-empty?